src/main/java/hudson/plugins/git/GitSCM.java

```diff
@@ -1325,14 +1332,38 @@ private void computeChangeLog(GitClient git, Revision revToBuild, TaskListener l
                 }
             }
 
+            if (context.getEnvironment().containsKey("CHANGE_TARGET")) {
+                listener.getLogger().println("Found CHANGE_TARGET in environment, considering it as origin.");
+                final String name = context.getEnvironment().get("CHANGE_TARGET");
+                git.fetch(getRepositories().get(0).getName(), new RefSpec("refs/heads/" + name + ":" + "refs/heads/" + name));
```

Comment:
This adds a duplicate `git fetch` into every multibranch pipeline, doesn't it?

We've had reports that `git fetch` can be expensive in large scale environments.  Is there any way to get the same information without the overhead of adding an additional fetch?

Reply:
I fully understand performance problems. Here, I think it is a now a full duplicated `fetch` ... I just fetch the remote branch recorded in the `CHANGE_TARGET` environment variable, because I need it locally to resolve it as a revision to stop changelog on it